crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -797,6 +797,63 @@ impl Persistence {
         ex.commit().await?;
         Ok(())
     }
+
+    pub async fn store_settlement_execution(
+        &self,
+        event: domain::settlement::Execution,
+    ) -> Result<(), DatabaseError> {
+        let mut ex = self.postgres.pool.acquire().await.context("acquire")?;
+        match event {
+            domain::settlement::Execution::Started {
+                auction_id,
+                solver,
```

Comment:
nit: Having the fields directly on the enum variant instead of a separate struct makes this function quite awkward IMO. If you use 2 structs instead you could have a very simple match on the enum that dispatches to 2 separate functions.

Ideally rust would allow functions to take enum variants instead of the whole enum to not require these helper struct but unfortunately that is not a thing.

Also having this enum probably is not super helpful anyway. You already have 2 separate functions for handling the DB logic and different logic in the run loop. So why not ditch the enum altogether and have 2 DB functions that take the 2 enum variants as structs directly?